mscore/musescore.cpp

```diff
@@ -587,6 +589,86 @@ void MuseScore::populateNoteInputMenu()
             }
       }
 
+//---------------------------------------------------------
+//   populateFileOperations
+//---------------------------------------------------------
+
+void MuseScore::populateFileOperations()
+      {
+      fileTools->clear();
+
+      if (qApp->layoutDirection() == Qt::LayoutDirection::LeftToRight) {
+            for (auto s : _fileOperationEntries) {
+                  if (!*s)
+                        fileTools->addSeparator();
+                  else
+                        fileTools->addWidget(new AccessibleToolButton(fileTools, getAction(s)));
+                  }
+            }
+      else {
+            _fileOperationEntries.reverse();
+            for (auto s : _fileOperationEntries) {
+                  if (!*s)
+                        fileTools->addSeparator();
+                  else
+                        fileTools->addWidget(new AccessibleToolButton(fileTools, getAction(s)));
+                  }
+            _fileOperationEntries.reverse();
+            }
+
+      // Currently not customizable in ToolbarEditor
+      fileTools->addSeparator();
+      mag = new MagBox;
+      connect(mag, SIGNAL(magChanged(MagIdx)), SLOT(magChanged(MagIdx)));
+      fileTools->addWidget(mag);
+
+      viewModeCombo = new QComboBox(this);
+#if defined(Q_OS_MAC)
+      viewModeCombo->setFocusPolicy(Qt::StrongFocus);
+#else
+      viewModeCombo->setFocusPolicy(Qt::TabFocus);
+#endif
+      viewModeCombo->setFixedHeight(preferences.getInt(PREF_UI_THEME_ICONHEIGHT) + 8);  // hack
+
+      viewModeCombo->setAccessibleName(tr("View Mode"));
+      viewModeCombo->addItem(tr("Page View"), int(LayoutMode::PAGE));
+      viewModeCombo->addItem(tr("Continuous View"), int(LayoutMode::LINE));
+      viewModeCombo->addItem(tr("Single Page"), int(LayoutMode::SYSTEM));
+
+      connect(viewModeCombo, SIGNAL(activated(int)), SLOT(switchLayoutMode(int)));
+      fileTools->addWidget(viewModeCombo);
+      }
+//---------------------------------------------------------
+//   populatePlaybackControls
+//---------------------------------------------------------
+
+void MuseScore::populatePlaybackControls()
+      {
+      transportTools->clear();
+
+      for (const auto s : _playbackControlEntries) {
+            if (!*s)
+                  transportTools->addSeparator();
+            else {
+                  if (QString(s) == "midi-on") {
+#ifdef HAS_MIDI
+                        transportTools->addWidget(new AccessibleToolButton(transportTools, getAction(s)));
+#endif
```

Comment:
doesn't this result in starting with a separator in cans of no MIDI? Wouldn't we need to skip that separator in an `#else` branch? The old code certainly only added the separator in the `#if` branch.

Reply:
In other apps I've used with customizable toolbars, buttons are buttons, separators are separators, and there is no connection between them.  Adding a button doesn't ever add a separator, or does deleting a button delete a separator, nor does moving a button move a separator.  Both are under the user's independent control.

Reply:
For midi not being there, maybe it should remove both the midi option and the separator on the first run and then let the user customize it from then on. Then it will work for the default workspaces and anything custom past that is the user’s choice.

Reply:
Now that whole section is no longer needed, as that string (and the separator) doesn't exist if HAS_MIDI is not #define'd